datasets/air_dialogue/air_dialogue.py

```diff
@@ -0,0 +1,298 @@
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, dev),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        # TODO: This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
+        # It is in charge of opening the given file and yielding (key, example) tuples from the dataset
+        # The key is not important, it's more here for legacy reason (legacy from tfds)
+
+        with open(filepath, encoding="utf-8") as f:
+            for id_, row in enumerate(f):
+                data = json.loads(row)
+                if self.config.name == "air_dialogue_data":
+
+                    intent = {
+                        "return_month": "" if "return_month" not in data["intent"] else data["intent"]["return_month"],
+                        "return_day": "" if "return_day" not in data["intent"] else data["intent"]["return_day"],
+                        "max_price": -1 if "max_price" not in data["intent"] else data["intent"]["max_price"],
+                        "departure_airport": ""
+                        if "departure_airport" not in data["intent"]
+                        else data["intent"]["departure_airport"],
+                        "max_connections": -1
+                        if "max_connections" not in data["intent"]
+                        else data["intent"]["max_connections"],
+                        "departure_day": ""
+                        if "departure_day" not in data["intent"]
+                        else data["intent"]["departure_day"],
+                        "goal": "" if "goal" not in data["intent"] else data["intent"]["goal"],
+                        "departure_month": ""
+                        if "departure_month" not in data["intent"]
+                        else data["intent"]["departure_month"],
+                        "name": "" if "name" not in data["intent"] else data["intent"]["name"],
+                        "return_airport": ""
+                        if "return_airport" not in data["intent"]
+                        else data["intent"]["return_airport"],
+                    }
+
+                    search_info = (
+                        []
+                        if "search_info" not in data
+                        else [
+                            {
+                                "button_name": "" if "button_name" not in search_info else search_info["button_name"],
+                                "field_name": "" if "field_name" not in search_info else search_info["field_name"],
+                                "field_value": "" if "field_value" not in search_info else search_info["field_value"],
+                                "timestmamp": -1 if "timestmamp" not in search_info else search_info["timestmamp"],
+                            }
```

Comment:
```suggestion
                            {
                                "button_name": search_info.get("button_name", ""),
                                "field_name": search_info.get("field_name", ""),
                                "field_value": search_info.get("field_value", ""),
                                "timestmamp": search_info["timestmamp"],
                            }
```

Reply:
here except for the `timestamp `(or `timestmamp`) other fields are optional. So we will require default checks.

Reply:
I changed to using `get` here.
It's equivalent to what you were doing but it's easier to read IMO